x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_details/agent_details_integrations.tsx

```diff
@@ -113,6 +119,30 @@ export const AgentDetailsIntegration: React.FunctionComponent<{
     );
   }, [agent, extensionView]);
 
+  const packageErrors = useMemo(() => {
+    const packageErrorUnits: FleetServerAgentComponentUnit[] = [];
+    if (!agent.components) {
+      return packageErrorUnits;
+    }
+
+    for (let i = 0; i < agent.components?.length; i++) {
+      if (agent.components[i].type === packagePolicy.package?.name) {
```

Comment:
@joeypoon  This is to only apply the errors of the current package, for example, `endpoint`. Do you know if there is only one component in the components array for each package (for example, only one component for `type==='endpoint'` ?  In that case, we could just use a `find` instead of a `for` loop

Reply:
fairly certain each package should be unique within the `components` array but waiting on verification.